Upgrades do_after and progress bars.

MrsTonedOne opened the pull request:
Fixes #14052

Fixes #14049

They will now always tick at 1ds, rather then by delay/numofticks.

Reworked my change with lastmove, SHOULD be fine now, as failed moves won't count.

do_mob and do_after moved to __helpers/mob.dm

Progress bar system made a datum and moved to datums/

Progress bars now have 20 icon states, or 5% resolution. 1 for each of the 20 pixels in the actual bar area.

Renamed progress bar icon file to........... progressbars

https://tgstation13.org/msoshit/progbar.flv

https://www.youtube.com/watch?v=ZaxuYun9ikA for you fucks who can't be assed to view a .flv file in a video player

:cl:
tweak: Progress bars now have 20 states instead of 5.
fix: Bump mining should be fixed.
tweak: Long actions now checks for movement/changed hands/etc more often to prevent edge cases.
tweak: Multiple progress bars will now stack properly, rather than fight each other for focus.
/:cl:

~~~I have a few more things to do, to fix bugs before this is ready.~~~ I'll do it in another PR, this is ready

MrsTonedOne committed:
Upgrades do_after and progress bars

Iamgoofball commented:
>flv

RemieRichards commented:
:+1:

phil235 commented on code/__HELPERS/mobs.dm:
```diff
@@ -139,4 +139,90 @@ Proc for attack log creation, because really why not
+			break
+		if(uninterruptible)
+			continue
+		if(user.loc != user_loc || user.last_move || target.loc != target_loc || target.last_move != null || user.get_active_hand() != holding || user.incapacitated() || user.lying )
```
why not use the same method as with do_after() and have `!target.last_move` instead of `target.last_move != null` ?

MrStonedOne replied:
must have missed it

MrsTonedOne committed:
Update mobs.dm

MrsTonedOne committed:
removes last_move checks from do_after/do_mob

it ticks at 1ds now, the cases where you could move and come back are rare, and require the server to be running at 21fps or higher and you have perfect lag and timing.
